File: app/src/main/java/com/sharpdroid/registroelettronico/Activities/AllLessonsWithDownloadActivity.java
package com.sharpdroid.registroelettronico.Activities;

import android.os.Bundle;
import android.support.design.widget.CoordinatorLayout;
import android.support.v4.widget.SwipeRefreshLayout;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;
import android.view.View;

import com.orm.SugarRecord;
import com.sharpdroid.registroelettronico.Adapters.AllLessonsAdapter;
import com.sharpdroid.registroelettronico.Databases.Entities.Lesson;
import com.sharpdroid.registroelettronico.Databases.Entities.Subject;
import com.sharpdroid.registroelettronico.R;
import com.sharpdroid.registroelettronico.Utils.Metodi;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class AllLessonsWithDownloadActivity extends AppCompatActivity implements SwipeRefreshLayout.OnRefreshListener {
    private final static String TAG = "Lessons";
    static int code;
    @BindView(R.id.recycler)
    RecyclerView mRecyclerView;
    @BindView(R.id.coordinator_layout)
    CoordinatorLayout mCoordinatorLayout;
    @BindView(R.id.swiperefresh)
    SwipeRefreshLayout mSwipeRefreshLayout;
    AllLessonsAdapter mRVAdapter;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_recycler_refresh_scrollbar);
        ButterKnife.bind(this);

        code = getIntent().getIntExtra("code", -1);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        if (getSupportActionBar() != null) {
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
            getSupportActionBar().setHomeAsUpIndicator(R.drawable.ic_close_black_24dp);
            getSupportActionBar().setDisplayShowHomeEnabled(true);
        }

        setTitle(Metodi.capitalizeEach(SugarRecord.findById(Subject.class, code).getDescription()));

        mRVAdapter = new AllLessonsAdapter(this);
        mRecyclerView.setLayoutManager(new LinearLayoutManager(this));
        mRecyclerView.setAdapter(mRVAdapter);

        mSwipeRefreshLayout.setVisibility(View.VISIBLE);//setRefreshing(true);
        mSwipeRefreshLayout.setOnRefreshListener(this);
        mSwipeRefreshLayout.setColorSchemeResources(
                R.color.bluematerial,
                R.color.redmaterial,
                R.color.greenmaterial,
                R.color.orangematerial);
        //UpdateLessons();
    }

    @Override
    protected void onResume() {
        super.onResume();

        bindLessonsCache();

    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) finish();
        return super.onOptionsItemSelected(item);
    }

    private void addLessons(List<Lesson> lessons) {
        if (!lessons.isEmpty()) {
            mRVAdapter.clear();
            mRVAdapter.addAll(lessons);
        }
    }

    private void bindLessonsCache() {
        addLessons(SugarRecord.findWithQuery(Lesson.class, "select * from LESSON where M_SUBJECT_ID=? ORDER BY M_DATE DESC", String.valueOf(code)));
    }


    @Override
    protected void onDestroy() {

        super.onDestroy();
    }

    @Override
    public void onRefresh() {
        mSwipeRefreshLayout.setRefreshing(false);
    }
}

File: app/src/main/java/com/sharpdroid/registroelettronico/activities/AllLessonsWithDownloadActivity.kt
package com.sharpdroid.registroelettronico.activities

import android.os.Bundle
import android.support.v4.widget.SwipeRefreshLayout
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import android.view.MenuItem
import android.view.View
import com.sharpdroid.registroelettronico.R
import com.sharpdroid.registroelettronico.adapters.AllLessonsAdapter
import com.sharpdroid.registroelettronico.database.pojos.LessonMini
import com.sharpdroid.registroelettronico.database.room.DatabaseHelper
import com.sharpdroid.registroelettronico.utils.Account
import com.sharpdroid.registroelettronico.utils.Metodi
import kotlinx.android.synthetic.main.activity_recycler_refresh_scrollbar.*
import kotlinx.android.synthetic.main.app_bar_main.*

class AllLessonsWithDownloadActivity : AppCompatActivity(), SwipeRefreshLayout.OnRefreshListener {
    private lateinit var mRVAdapter: AllLessonsAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_recycler_refresh_scrollbar)

        val code = intent.getIntExtra("code", -1)

        setSupportActionBar(toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.setHomeAsUpIndicator(R.drawable.ic_close_black_24dp)
        supportActionBar?.setDisplayShowHomeEnabled(true)

        mRVAdapter = AllLessonsAdapter()

        title = "Lezioni"

        with(recycler) {
            layoutManager = LinearLayoutManager(this@AllLessonsWithDownloadActivity)
            adapter = mRVAdapter
        }

        with(swiperefresh) {
            visibility = View.VISIBLE
            setOnRefreshListener(this@AllLessonsWithDownloadActivity)
            setColorSchemeResources(
                    R.color.bluematerial,
                    R.color.redmaterial,
                    R.color.greenmaterial,
                    R.color.orangematerial)
        }

        addLessons(DatabaseHelper.database.lessonsDao().loadLessons(code.toLong(), Account.with(this).user))
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) finish()
        return super.onOptionsItemSelected(item)
    }

    private fun addLessons(lessons: List<LessonMini>) {
        mRVAdapter.clear()
        mRVAdapter.addAll(lessons)
    }

    override fun onRefresh() {
        Metodi.updateLessons(this)
    }

}
